core/src/main/scala/org/apache/spark/api/r/RBackendHandler.scala

```diff
@@ -148,6 +151,9 @@ private[r] class RBackendHandler(server: RBackend)
       case e: Exception =>
         logError(s"$methodName on $objId failed", e)
         writeInt(dos, -1)
+        // Writing the error message of the cause for the exception. This will be returned
+        // to user in the R process.
+        writeString(dos, e.getCause.getMessage)
```

Comment:
would it make sense to write the stack too? often time it is much more useful to have the call stack in additional to the exception message